packages/lesswrong/server/utils/contributorsFieldHelper.ts

```diff
@@ -0,0 +1,70 @@
+import { getContributorsList, ContributorWithStats } from './contributorsUtil';
+import { loadByIds } from '@/lib/loaders';
+import { accessFilterMultiple } from '@/lib/utils/schemaUtils';
+import keyBy from 'lodash/keyBy';
+import orderBy from 'lodash/orderBy';
+import take from 'lodash/take';
+
+interface ContributorsFieldOptions {
+  collectionName: 'Tags' | 'MultiDocuments';
+  fieldName: string;
+  graphQLType: string;
+  arguments?: string;
+}
+
+export function contributorsField(options: ContributorsFieldOptions) {
+  const { collectionName, fieldName, graphQLType, arguments: graphqlArgs } = options;
+
+  return {
+    resolveAs: {
+      type: graphQLType,
+      arguments: graphqlArgs || 'limit: Int, version: String',
```

Comment:
Not obvious to me what the right thing to do here is, but I think dynamic graphql arguments are a bad idea.  When I was working on the initial prototype I ran into problems related to the fact that if you have fragments which include resolver fields with graphql variables passed to them, you then have to include those in the query you send from `useMulti` via `extraVariables`/`extraVariablesValues`.  It wasn't obvious at first because the variable names were the same for the tag field and the multidocuments field, but if the variable names are the same, it ends up using the same value in both cases.

The other headache is that you then have to figure out what variable names you're passing in here so that you type the `args` in the `resolver` correctly.  Right now you're typing it as `{ limit, version }`, which is only correct for tags, and not correct for multidocuments (and so the versioning functionality won't work).

Reply:
Possibly we just require the caller of `contributorsField` to pass in the intended field names for both the `limit` and `version` variables?

Reply:
Actually, the first set of concerns... idk, maybe not an issue.  But empirically we aren't passing through a limit or version in any of the fragments for multidocuments...

Reply:
Gonna be honest and say I don't fully follow this, but we can discuss